core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala

```diff
@@ -351,6 +351,10 @@ class ContextCleanerSuite extends ContextCleanerSuiteBase {
       case _ => false
     }, askStorageEndpoints = true).isEmpty)
   }
+
+  test("SPARK-33000: shutdown cleans up checkpointed data when configured to do so") {
+    // TODO: How do I write a test for this?
+  }
```

Comment:
I am not sure what would be an appropriate test to add here. I can directly call the `cleanupOnShutdown()` method, but that doesn't prove that Spark will do the right thing when it shuts down.

Ideally, I would write some sort of acceptance test that launches a tiny Spark application and checks the checkpoint directory afterwards. But that seems very bulky and I don't know if there is an existing pattern in the codebase that I can follow.

Reply:
We have a couple of thrift tests that do something like that. I have to admit it is not pretty.